### PR TITLE
Chartdata collection refactor

### PR DESCRIPTION
**Pull request by jjatie**
Refactored logic where possible to take advantage of `ChartData`'s `Collection` conformance

### CODE REVIEW

**Comment by jjatie** [comment]
This is in conjunction with #3023

File: ChartsDemo/Swift/Demos/BubbleChartViewController.swift [hunk 1]
```diff
@@ -104,7 +104,7 @@ class BubbleChartViewController: DemoBaseViewController {
         set3.setColor(ChartColorTemplates.colorful()[2], alpha: 0.5)
         set3.drawValuesEnabled = true
         
-        let data = BubbleChartData(dataSets: [set1, set2, set3])
+        let data = [set1, set2, set3] as BubbleChartData
```
Comment (liuxuan30) [inline ChartsDemo/Swift/Demos/BubbleChartViewController.swift]: may I know which protocol method enables this convenient conversion?

File: ChartsDemo/Swift/DemoBaseViewController.swift [hunk 2]
```diff
@@ -117,13 +117,13 @@ class DemoBaseViewController: UIViewController, ChartViewDelegate {
     func handleOption(_ option: Option, forChartView chartView: ChartViewBase) {
         switch option {
         case .toggleValues:
-            for set in chartView.data!.dataSets {
+            for set in chartView.data! {
```
Comment (liuxuan30) [inline ChartsDemo/Swift/DemoBaseViewController.swift]: I had a feeling that `for set in data` is less natural than `for set in data.dataSets`, what you think? Maybe need time to adapt?
So in the future, we will deprecate `data.dataSet`?

File: ChartsDemo/Swift/Demos/LineChart1ViewController.swift [hunk 3]
```diff
@@ -144,31 +144,31 @@ class LineChart1ViewController: DemoBaseViewController {
     override func optionTapped(_ option: Option) {
         switch option {
         case .toggleFilled:
-            for set in chartView.data!.dataSets as! [LineChartDataSet] {
+            for set in chartView.data as! LineChartData {
```
Comment (liuxuan30) [inline ChartsDemo/Swift/Demos/LineChart1ViewController.swift]: want to know which method enables this as well.
And, it seems weird that a `set` object is a `ChartData` type? what you think

File: Source/Charts/Highlight/ChartHighlighter.swift [hunk 4]
```diff
@@ -73,17 +73,11 @@ open class ChartHighlighter : NSObject, Highlighter
...
         
-        for i in 0 ..< data.dataSetCount
-        {
-            guard
-                let dataSet = data.getDataSetByIndex(i),
-                dataSet.isHighlightEnabled      // don't include datasets that cannot be highlighted
-                else { continue }
-            
-
+        for (i, set) in zip(data.indices, data) where set.isHighlightEnabled
```
Comment (liuxuan30) [inline Source/Charts/Highlight/ChartHighlighter.swift]: I like this advanced use. Does this come with the collection adoption?

Comment (jjatie) [reply Source/Charts/Highlight/ChartHighlighter.swift]: `for d in data` comes from `Sequence` adoption which `Collection` inherits. This also allows the zip(:) function to be used.

File: ChartsDemo/Swift/Demos/LineChart1ViewController.swift (hunk 3 above)
Comment (jjatie) [reply ChartsDemo/Swift/Demos/LineChart1ViewController.swift]: See comment below

File: ChartsDemo/Swift/Demos/BubbleChartViewController.swift (hunk 1 above)
Comment (jjatie) [reply ChartsDemo/Swift/Demos/BubbleChartViewController.swift]: `ExpressibleByArrayLiteral`

File: ChartsDemo/Swift/DemoBaseViewController.swift (hunk 2 above)
Comment (jjatie) [reply ChartsDemo/Swift/DemoBaseViewController.swift]: I think it's just new to you. We need to continue to support `data.dataSet` as long as we keep Obj-C compatibility.

**Comment by liuxuan30** [comment]
I ran twice, still failure.

**Comment by jjatie** [comment]
Any chance you can add an approval or additional review before you can go to bed? Travis CI is finally getting better and I could merge during my day. BTW my time zone is GMT-0400 hours

File: ChartsDemo/Swift/Demos/LineChart1ViewController.swift (hunk 3 above)
Comment (liuxuan30) [reply ChartsDemo/Swift/Demos/LineChart1ViewController.swift]: @jjatie `for set in chartView.data as! LineChartData` here I'm confused, as `set` implies a dataset class, while it casts to a chartdata class. is the name good here?

**Comment by liuxuan30** [comment]
and when I wake up and still pending LOL.
The only question is about some names. see comments above that you didn't answered. Otherwise, it seems good.

File: ChartsDemo/Swift/Demos/LineChart1ViewController.swift (hunk 3 above)
Comment (liuxuan30) [reply ChartsDemo/Swift/Demos/LineChart1ViewController.swift]: oh wait. Seems I asked a wrong question?
it's `chartView.data` being casted to `LineChartData`?

**Comment by liuxuan30** [comment]
Travis macOS builds have trouble again. Expect delay

**Pull request merged**